<launch>
  <!-- Start full simulation + controllers -->
  <include file="$(find sentry_chassis_controller)/launch/sentry_pid_test_fixed.launch"/>

  <!-- Start forward kinematics node to compute odometry from joint_states -->
  <node pkg="sentry_chassis_controller" type="forward_kinematics" name="forward_kinematics" output="screen">
    <param name="wheel_base" value="0.36"/>
    <param name="wheel_track" value="0.36"/>
    <param name="wheel_radius" value="0.05"/>
  </node>

  <!-- C++ keyboard teleop with spinning top mode support -->
  <node pkg="sentry_chassis_controller" type="sentry_control_key" name="sentry_control_key" output="screen">
    <param name="cmd_vel_topic" value="/cmd_vel"/>
    <!-- 优化为小陀螺模式调整的参数 -->
    <param name="walk_vel" value="0.8"/>     <!-- 适中的平移速度 -->
    <param name="run_vel" value="1.5"/>      <!-- 加速模式速度 -->
    <param name="yaw_rate" value="2.0"/>     <!-- 稍快的自转速度 -->
    <param name="yaw_rate_run" value="3.0"/> <!-- 自转加速模式 -->
    <param name="hz" value="30"/>            <!-- 更高频率以获得更平滑的控制 -->
  </node>

</launch>